File: lecture_006(2DArray)/l_011_addmatrix.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<vector<int>> &arr) {
     for(int row=0; row<arr.size(); row++)  {
        for(int col=0; col<arr[0].size(); col++) {
            cout<<arr[row][col]<<" ";
        }
        cout<<endl;
    }
}
void input(vector<vector<int>> &arr) {
    for(int row=0; row<arr.size(); row++)  {
        for(int col=0; col<arr[0].size(); col++) {
            cin>>arr[row][col];
        }
    }
    display(arr);
}
void addMatrix(vector<vector<int>> &arr, vector<vector<int>> &arr1) {
     for(int row=0; row<arr.size(); row++)  {
        for(int col=0; col<arr[0].size(); col++) {
            arr[row][col] += arr1[row][col];
        }
    }
    display(arr);
}    
int main(int args, char**argvs) {
    int n,m;
    cin>>n>>m;
    vector<vector<int>>arr(n,vector<int>(m,0));
    vector<vector<int>>arr1(n,vector<int>(m,0));
    input(arr);
    input(arr1);
    addMatrix(arr,arr1);
}

File: lecture_006(2DArray)/l_011_addmatrix.java
import java.util.Scanner;
public class l_011_addmatrix {
	public static Scanner scn = new Scanner(System.in);
	public static void main(String a[]) {
		int n = scn.nextInt();
		int m = scn.nextInt();
		int[][] arr = new int[n][m];
		input(arr);
		int[][] brr = new int[n][m];
		input(brr);
		int[][] crr = new int[n][m];
		addmatrix(arr,brr,crr);
	}
	public static void display(int[][] arr) {
        for(int i=0; i<arr.length; i++) {
            for(int j=0; j<arr[0].length; j++) {
                System.out.print(arr[i][j]+" ");
            }
            System.out.println();
        }
    }
    public static void input(int[][] arr) {
        for(int i=0; i<arr.length; i++) {
            for(int j=0; j<arr[0].length; j++) {
                arr[i][j] = scn.nextInt();
            }
        }
        display(arr);
    }
    public static void addmatrix(int[][] arr, int[][] brr, int[][] crr) {
    	for(int i=0; i<crr.length; i++) {
    		for(int j=0; j<crr[0].length; j++) {
    			crr[i][j] = arr[i][j] + brr[i][j];
    		}
    	}
    	display(crr);
    }
}